pandas/core/groupby/groupby.py

```diff
@@ -2463,6 +2463,22 @@ def ohlc(self) -> DataFrame:
     @doc(DataFrame.describe)
     def describe(self, **kwargs):
         with self._group_selection_context():
+            if len(self._selected_obj) == 0:
+                if self._selected_obj.ndim == 1:
+                    return self._selected_obj.describe(**kwargs).to_frame().T.iloc[:0]
+                else:
+                    selected_obj_describe = self._selected_obj.describe(**kwargs).T
+                    selected_obj_describe_columns = selected_obj_describe.columns
+                    selected_obj_describe_indexes = selected_obj_describe.index
+                    return DataFrame(
+                        columns=MultiIndex.from_product(
+                            [
+                                selected_obj_describe_indexes,
+                                selected_obj_describe_columns,
+                            ]
+                        )
+                    )
```

Comment:
My original comment in the DataFrame case had a piece missing, which is why I suspect you didn't use it :-). It should have been

    self._selected_obj.describe().unstack().to_frame().T.iloc[:0]

What you have looks good except the dtypes come out incorrect. But to get the consistent dtype is a bit tricky, it depends on the input dtype:

```
df = pd.DataFrame({'a': [1, 1, 3], 'b': [4, 5, 6]})
print(df.groupby('a').describe())
df = df.astype(object)
print(df.groupby('a').describe())

      b                                          
  count mean       std  min   25%  50%   75%  max
a                                                
1   2.0  4.5  0.707107  4.0  4.25  4.5  4.75  5.0
3   1.0  6.0       NaN  6.0  6.00  6.0  6.00  6.0
      b                
  count unique top freq
a                      
1     2      2   4    1
3     1      1   6    1
```

In the first case, count is a float and in the 2nd case, count is an integer. This is why I recommend the above method.

Reply:
ok. Have updated.